libraries/default.rb

```diff
@@ -0,0 +1,69 @@
+def collect_github_repositories(github_token, org)
+  if ::ObjectSpace.const_defined?('Chef')
+    # Chef exists! Lets make sure Octokit is installed.
+    chef_gem 'octokit'
+  end
+
+  require 'octokit'
+
+  github = Octokit::Client.new(access_token: github_token)
+  github.auto_paginate = true
+
+  github.org_repos(org).map(&:name)
+end
+
+def public_address
+  node.fetch('cloud', {}).fetch('public_ipv4', nil) || node['fqdn']
+end
```

Comment:
I think this should be put in `libraries/helpers.rb` and these other github functions should be pushed to `libraries/github.rb`